x/expiration/client/cli/query.go

```diff
@@ -0,0 +1,233 @@
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/version"
+
+	"github.com/provenance-io/provenance/x/expiration/types"
+)
+
+// GetQueryCmd is the top-level command for expiration CLI queries
+func GetQueryCmd() *cobra.Command {
+	queryCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Aliases:                    []string{"exp"},
+		Short:                      "Querying commands for the expiration module",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+	queryCmd.AddCommand(
+		GetQueryParamsCmd(),
+		GetExpirationCmd(),
+		GetAllExpirationsCmd(),
+		GetAllExpirationsByOwnerCmd(),
+	)
+
+	return queryCmd
+}
+
+// GetQueryParamsCmd is the CLI command for expiration parameter querying
+func GetQueryParamsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "params",
+		Aliases: []string{"p"},
+		Short:   "Query the current expiration parameters",
+		Args:    cobra.NoArgs,
+		Example: fmt.Sprintf(`$ %s query expiration params`, version.AppName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			queryClient := types.NewQueryClient(clientCtx)
+			res, err := queryClient.Params(context.Background(), &types.QueryParamsRequest{})
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(&res.Params)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// GetExpirationCmd is the CLI command for listing a specific expiration
+func GetExpirationCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "get [module-asset-id]",
+		Aliases: []string{"g"},
+		Short:   "Query expiration for a module asset",
+		Args:    cobra.ExactArgs(1),
+		Example: fmt.Sprintf(`$ %s query expiration get pb1skjwj5whet0lpe65qaq4rpq03hjxlwd9nf39lk`, version.AppName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			queryClient := types.NewQueryClient(clientCtx)
+			req := &types.QueryExpirationRequest{ModuleAssetId: strings.TrimSpace(args[0])}
+			res, err := queryClient.Expiration(context.Background(), req)
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// GetAllExpirationsCmd is the CLI command for listing all expirations
+func GetAllExpirationsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "all",
+		Aliases: []string{"a"},
+		Short:   "Query all expirations",
+		Args:    cobra.NoArgs,
+		Example: strings.TrimSpace(
+			fmt.Sprintf(`
+				$ %[1]s query expiration all
+				$ %[1]s query expiration all --page=2 --limit=100
+				`,
+				version.AppName,
+			)),
```

Comment:
Because this string is inside tics, all characters are included. That means that the initial newline after the first tic is in the string as well as the 4 leading tabs on each line.

```suggestion
			fmt.Sprintf(`$ %[1]s query expiration all
$ %[1]s query expiration all --page=2 --limit=100`,
				version.AppName,
			)),
```

Also, maybe consider creating a variable to hold the string `version.AppName + " query expiration"`, and providing that in all these commands instead of `version.AppName` and repeating those next two pieces each time.